### PR TITLE
NAS-124674 / 24.04 / Fixes for SMART tests

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Testing: see ticket comments

### CODE REVIEW

File: src/app/helptext/storage/disks/disks.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ export default {
   fieldset_disk: T('General'),
   fieldset_powermgmt: T('Power Management'),
   fieldset_temperature: T('Temperature Alerts'),
-  fieldset_smartsed: T('S.M.A.R.T./SED'),
+  fieldset_smart: T('S.M.A.R.T.'),
```
Comment (undsoft) [inline src/app/helptext/storage/disks/disks.ts:10]: We can remove `T()`. The string will be the same in all languages.

File: src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts [hunk 2]
```diff
@@ -99,6 +100,10 @@ export class ManualTestDialogComponent {
             next: (result) => {
               if (result?.progress) {
                 this.progressTotalPercent = result.progress.percent / 100;
+
+                if (this.progressTotalPercent === 100) {
+                  this.dialogRef.close();
```
Comment (undsoft) [inline src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts:105]: This doesn't look pretty. Wouldn't the job complete and dialog be closed on its own?

File: src/app/modules/entity/entity-table/entity-table.component.ts [hunk 3]
```diff
@@ -1178,6 +1180,10 @@ export class EntityTableComponent<Row extends SomeRow = SomeRow> implements OnIn
     return this.conf.config.multiSelect ? this.currentColumns[1].prop : this.currentColumns[0].prop;
   }
 
+  getRowValue(value: boolean | string): string {
+    return typeof value === 'boolean' ? this.yesNoPipe.transform(value) : value;
```
Comment (undsoft) [inline src/app/modules/entity/entity-table/entity-table.component.ts:1184]: I'm worried about unintended consequences of editing a core component like this.
I'd rather we use local transformations in disks component.

**Review by undsoft — CHANGES_REQUESTED** [review]
1. Still says `Enable S.M.A.R.T.: N/A` in details.
2. I'd rather we wrap new sections and not have 3 sections in the same row.

File: src/app/modules/entity/entity-table/entity-table.component.ts (hunk 3 above)
Comment (AlexKarpov98) [reply src/app/modules/entity/entity-table/entity-table.component.ts:1184]: @undsoft - you mean modify it from boolean to string in disks component?

File: src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts:105]: > This doesn't look pretty. Wouldn't the job complete and dialog be closed on its own?

I Just added extra check to close it once progress is 100%

File: src/app/modules/entity/entity-table/entity-table.component.ts (hunk 3 above)
Comment (undsoft) [reply src/app/modules/entity/entity-table/entity-table.component.ts:1184]: Yes.

File: src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts (hunk 2 above)
Comment (undsoft) [reply src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts:105]: Is it necessary?

Comment (AlexKarpov98) [reply src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts:105]: > Is it necessary?

nope, at all

Comment (AlexKarpov98) [reply src/app/pages/storage/modules/disks/components/manual-test-dialog/manual-test-dialog.component.ts:105]: I'll remove it.

File: src/app/modules/entity/entity-table/entity-table.component.ts (hunk 3 above)
Comment (AlexKarpov98) [reply src/app/modules/entity/entity-table/entity-table.component.ts:1184]: We need more tables to be refactored to ix-table2 😄

**Pull request merged**